tests/SetCookieTest.php

```diff
@@ -44,58 +47,64 @@ public function test_it_can_be_added_to_a_psr_response()
+        self::assertSame('name=value; path=/path/', $cookie->toHeaderValue());
+
+        $cookie = new SetCookie('name', 'value', 0, '0'); // falsey path value
+        self::assertSame('name=value; path=0', $cookie->toHeaderValue());
 
         $cookie = new SetCookie('name', 'value', 0, '/path/', 'domain.tld');
-        $this->assertEquals('name=value; path=/path/; domain=domain.tld', $cookie->toHeaderValue());
+        self::assertSame('name=value; path=/path/; domain=domain.tld', $cookie->toHeaderValue());
 
         $cookie = new SetCookie('name', 'value', 0, '/path/', 'domain.tld', true);
-        $this->assertEquals('name=value; path=/path/; domain=domain.tld; secure', $cookie->toHeaderValue());
+        self::assertSame('name=value; path=/path/; domain=domain.tld; secure', $cookie->toHeaderValue());
 
         $cookie = new SetCookie('name', 'value', 0, '/path/', 'domain.tld', true, true);
-        $this->assertEquals('name=value; path=/path/; domain=domain.tld; secure; httponly', $cookie->toHeaderValue());
+        self::assertSame('name=value; path=/path/; domain=domain.tld; secure; httponly', $cookie->toHeaderValue());
 
         $cookie = new SetCookie('name', 'value', 1466459967, '', '', true, true);
-        $this->assertEquals('name=value; expires=Mon, 20 Jun 2016 21:59:27 GMT; secure; httponly', $cookie->toHeaderValue());
+        self::assertSame('name=value; expires=Mon, 20 Jun 2016 21:59:27 GMT; secure; httponly', $cookie->toHeaderValue());
 
         $cookie = new SetCookie('name', 'value', 0, '/path/', 'domain.tld', true, true, 'strict');
-        $this->assertEquals('name=value; path=/path/; domain=domain.tld; secure; httponly; samesite=strict', $cookie->toHeaderValue());
+        self::assertSame('name=value; path=/path/; domain=domain.tld; secure; httponly; samesite=strict', $cookie->toHeaderValue());
 
         $cookie = SetCookie::thatDeletesCookie('name');
         $expected = sprintf('name=deleted; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, 1));
-        $this->assertEquals($expected, $cookie->toHeaderValue());
+        self::assertSame($expected, $cookie->toHeaderValue());
 
         $now = new DateTimeImmutable();
         $cookie = SetCookie::thatExpires('name', 'value', $now);
         $timestamp = (int) $now->format('U');
         $expected = sprintf('name=value; expires=%s', gmdate(self::$HTTP_DATE_FORMAT, $timestamp));
-        $this->assertEquals($expected, $cookie->toHeaderValue());
+        self::assertSame($expected, $cookie->toHeaderValue());
 
         $cookie = SetCookie::thatStaysForever('name', 'value', '/path/', 'domain.tld');
         $expiresInFiveYear = time() + 5 * 365 * 3600 * 24;
         $expected = sprintf('name=value; expires=%s; path=/path/; domain=domain.tld', gmdate(self::$HTTP_DATE_FORMAT, $expiresInFiveYear));
-        $this->assertEquals($expected, $cookie->toHeaderValue());
+        self::assertSame($expected, $cookie->toHeaderValue());
```

Comment:
I know these are statically defined but I prefer to keep the $this syntax :))

As this aligns with docs from PHPUnit.

> A common question, especially from developers new to PHPUnit, is whether using $this->assertTrue() or self::assertTrue(), for instance, is “the right way” to invoke an assertion. The short answer is: there is no right way. And there is no wrong way, either. It is a matter of personal preference.

https://docs.phpunit.de/en/10.5/assertions.html

Reply:
Well, the methods are defined as static, so you should call them as static - some code analysis tools will yell at you otherwise (you could still go for `$this::assertSame()`).